### DISCUSSION TITLE
When using dbx, how to update a multiple select field?

### DISCUSSION THREAD

**Discussion by rangzen** [pocketbase/pocketbase#2893] [Q&A] [answered]
If I use a string like "a,b,c" it creates a new string value as "a,b,c". If I send a slice, I have: "sql: converting argument $5 type: unsupported type []string, a slice of string".

Here values is a []string.

```go
_, err = r.builder.Update("tableName",
    dbx.Params{
        [...]
        "values":   values,
        [...]
    },
    dbx.HashExp{"key": key},
).Execute()```

**Comment by ganigeorgiev** [comment, marked as answer]
You need to create a Go type that implements `sql.Scanner` and `driver.Valuer` interfaces to scan/export slice values (_or any other non-primitive data_).

Alternatively, you can use the builtin [`types.JsonArray`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.8/tools/types#JsonArray): `types.JsonArray[string]{"a", "b", "c"}`

**Reply by ganigeorgiev** [reply]
_btw. if you use `models.Record` and its `Set("values", values)` helper it will normalize the field value for you._

**Reply by rangzen** [reply]
Write perfectly with values as []string and `jsonArray := types.JsonArray[string](values)`.
Thank you so much, I continue exploring :)
